pkg/controller/opentelemetrycollector/reconcile_service_test.go

```diff
@@ -145,3 +148,169 @@ func TestDeleteExtraService(t *testing.T) {
 	assert.Error(t, err) // not found
 	assert.Nil(t, persisted)
 }
+
+func TestParsePortFromEndpoint(t *testing.T) {
+	for _, tt := range []struct {
+		endpoint      string
+		expected      int32
+		errorExpected bool
+	}{
+		// prepare
+		{"http://localhost:1234", 1234, false},
+		{"0.0.0.0:1234", 1234, false},
+		{":1234", 1234, false},
+		{"no-port", 0, true},
+	} {
+
+		// test
+		val, err := portFromEndpoint(tt.endpoint)
+
+		// verify
+		if tt.errorExpected {
+			assert.Error(t, err)
+		} else {
+			assert.NoError(t, err)
+		}
+
+		assert.Equal(t, tt.expected, val, "wrong port from endpoint %s: %d", tt.endpoint, val)
+	}
+}
+
+func TestExtractPortsFromConfig(t *testing.T) {
+	// prepare
+	configStr := `receivers:
+  examplereceiver:
+    endpoint: "0.0.0.0:12345"
+  examplereceiver/settings:
+    endpoint: "0.0.0.0:12346"
+  examplereceiver/invalid-ignored:
+    endpoint: "0.0.0.0"
+  examplereceiver/invalid-not-number:
+    endpoint: "0.0.0.0:not-number"
+  examplereceiver/without-endpoint:
+    notendpoint: "0.0.0.0:12347"
+  jaeger:
+    grpc:
+    thrift_compact:
+    thrift_binary:
+      endpoint: 0.0.0.0:6833
+  jaeger/custom:
+    thrift_http:
+      endpoint: 0.0.0.0:15268
+`
+
+	// test
+	ports, err := ports(logger, configStr)
+
+	// verify
+	assert.NoError(t, err)
+	assert.Len(t, ports, 6)
+
+	expectedPorts := map[int32]bool{}
```

Comment:
Wouldn't it be better to define a map of expected names to ports - and then build a map from the returned ports and do a comparison?

Reply:
So that not only their existence is checked, but also whether the names are linked to the right ports?

Reply:
Yes

Reply:
I think this is addressed in the last commit.